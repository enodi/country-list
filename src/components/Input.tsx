import React from "react";

type InputProps = {
    placeholder: string;
    type: string;
    handleOnChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
};

const Input: React.FunctionComponent<InputProps> = ({
    placeholder,
    type,
    handleOnChange,
}) => (
    <input
        className="input"
        type={type}
        onChange={(event) => handleOnChange(event)}
        placeholder={placeholder}
    />
);

export default Input;
