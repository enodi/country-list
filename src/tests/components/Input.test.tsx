import React from "react";
import { shallow } from "enzyme";

import Input from "../../components/Input";

describe("Input Component", () => {
    let wrapper: any, handleOnChange: () => void;
    beforeEach(() => {
        handleOnChange = jest.fn();
        wrapper = shallow(
            <Input
                placeholder="Search"
                handleOnChange={handleOnChange}
                type="text"
            />
        );
    });

    it("should render Input component correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("should call handleOnChange on input change", () => {
        wrapper.find("input").simulate("change");
        expect(handleOnChange).toHaveBeenCalled();
    });
});
