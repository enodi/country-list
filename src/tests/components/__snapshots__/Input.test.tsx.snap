// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render Input component correctly 1`] = `
<input
  className="input"
  onChange={[Function]}
  placeholder="Search"
  type="text"
/>
`;
